lib/active_model_serializers/adapter/json_api.rb

```diff
@@ -234,6 +235,12 @@ def resource_objects_for(serializers)
         [@primary, @included]
       end
 
+      def paginate?
+        serialization_context &&
+          serializer.respond_to?(:paginated?) &&
```

Comment:
these two lines could be replaced with a `try` -- unless the team is against trys for some reason